service/bfg/bfg.go

```diff
@@ -1198,16 +1275,35 @@ func (s *Server) handleBtcFinalityByKeystonesRequest(ctx context.Context, bfkr *
 	}, nil
 }
 
-func (s *Server) handleL2KeystonesRequest(ctx context.Context, l2kr *bfgapi.L2KeystonesRequest) (any, error) {
-	log.Tracef("handleL2KeystonesRequest")
-	defer log.Tracef("handleL2KeystonesRequest exit")
+func (s *Server) getL2KeystonesCache() []hemi.L2Keystone {
+	log.Tracef("getL2KeystonesCache")
+	defer log.Tracef("getL2KeystonesCache exit")
+
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	if s.l2keystonesCache == nil {
+		return []hemi.L2Keystone{}
+	}
+
+	results := []hemi.L2Keystone{}
+	for _, v := range s.l2keystonesCache {
+		results = append(results, v)
+	}
+
+	return results
+}
+
+func (s *Server) refreshL2KeystoneCache(ctx context.Context) {
+	log.Tracef("refreshL2KeystoneCache")
+	defer log.Tracef("refreshL2KeystoneCache exit")
+
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
 
-	results, err := s.db.L2KeystonesMostRecentN(ctx, uint32(l2kr.NumL2Keystones))
+	results, err := s.db.L2KeystonesMostRecentN(ctx, 100)
 	if err != nil {
```

Comment:
This error check no longer returns, is that intentional?

Reply:
nice catch, just a mistake